duktape/src/main/jni/java/JavaType.cpp

```diff
@@ -135,11 +164,41 @@ struct Boolean : public Primitive {
     return value;
   }
 
+  jarray popArray(duk_context* ctx, JNIEnv* env, uint32_t count, bool inScript) const {
+    jbooleanArray array = env->NewBooleanArray(count);
+    for (auto i = count; i > 0u; --i) {
+      const auto value = pop(ctx, env, inScript).z;
+      env->SetBooleanArrayRegion(array, i - 1, 1, &value);
+    }
+    return array;
+  }
+
   duk_ret_t push(duk_context* ctx, JNIEnv*, const jvalue& value) const override {
     duk_push_boolean(ctx, value.z == JNI_TRUE);
     return 1;
   }
 
+  duk_ret_t pushArray(duk_context* ctx, JNIEnv* env, const jarray& values) const {
+    const auto size = env->GetArrayLength(values);
+    if (size == 0) {
+      return 0;
+    }
+    jvalue element;
+    jboolean* elements = env->GetBooleanArrayElements(static_cast<jbooleanArray>(values), nullptr);
+    for (jsize i = 0; i < size; ++i) {
+      element.z = elements[i];
+      try {
+        push(ctx, env, element);
+      } catch (std::invalid_argument& e) {
+        env->ReleaseBooleanArrayElements(static_cast<jbooleanArray>(values), elements, JNI_ABORT);
+        duk_pop_n(ctx, i);
+        throw e;
+      }
+    }
+    env->ReleaseBooleanArrayElements(static_cast<jbooleanArray>(values), elements, JNI_ABORT);
```

Comment:
I *should* write an RAII wrapper, but doing the release in two places seemed like less typing.